config/boot.rb

```diff
@@ -1,3 +1,14 @@
+# Set up gems listed in the Gemfile.
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile', __FILE__)
 
 require 'bundler/setup' # Set up gems listed in the Gemfile.
+
+# load dotenv files before "before_configuration" callback
+require File.expand_path('../configuration', __FILE__)
+
+# global instance to access and use
+AppConfig = Configuration.new
```

Comment:
The `AppConfig` variable doesn't follow typical Ruby naming conventions:

https://github.com/bbatsov/ruby-style-guide#naming

They recommend camel case for `Class` or `Module` only.

Reply:
I'd have to refresh my visual picture of the ruby object model, but I think that doing:

```
AppConfig = Configuration.new
```

is functionally equivalent to creating AppConfig class and then specifying all the config options as class methods in a `class << self do` block.



So this particular case is a little grey.  I introduced it in 6ac2d861093ef9b5e281e28fb40986895e34e5f0:

Its the first idea proposed in
https://8thlight.com/blog/josh-cheek/2012/10/20/implementing-and-testing-the-singleton-pattern-in-ruby.html

The idea is you want to use the Singleton pattern but you don't want the complexity of the Singleton pattern and you want the class you are making a singleton of easy to test.

Reply:
In otherwords, here I'm defining a new class called AppConfig.

Reply:
We discussed this offline, I am fine with this since it makes testing easier.